src/corehost/src/main.cpp

```diff
@@ -8,30 +8,53 @@
 #include "utils.h"
 #include "coreclr.h"
 
-void get_tpafile_path(const pal::string_t& app_base, const pal::string_t& app_name, pal::string_t& tpapath)
+void get_tpafile_path(const pal::string_t& app_base, const pal::string_t& app_name, pal::string_t* tpapath)
 {
-    tpapath.reserve(app_base.length() + app_name.length() + 5);
+    tpapath->reserve(app_base.length() + app_name.length() + 5);
```

Comment:
You should have a comment explaining why we have a "5". Also, shouldn't it be 6 since 5 are from ".deps" and 1 for DIR_SEPARATOR? Also, pal::string_t accounts for NULL?